server/server.js

```diff
@@ -18,48 +20,67 @@ var secret = require('./secret.js');
 
 app.use(bodyParser.json());
 
-app.use(session({
+var sessionMiddleware = session({
+  // should make session persist even if server crashes
+  store: new MongoStore({ mongooseConnection: db.connection }),
   secret: secret,
   resave: true,
   saveUninitialized: true
-}));
+})
+app.use(sessionMiddleware);
 
 app.use("/", express.static(__dirname + '/../client-web'));
 //new internal dependencies
 var router = require('./routes.js');
 
+//mount middleware to io request, now we have access to socket.request.session
+io.use(function(socket,next){
+  sessionMiddleware(socket.request,socket.request.res,next)
+})
 
 //***************** Sockets *******************
 //listen for connection event for incoming sockets
 //store all users that want to find a kwiky
 io.on('connection',function(socket){
-  console.log('Socket '+ socket.id +' connected.');
+  // console.log('Socket '+ socket.id +' connected.');
+  console.log('connection',socket.request.session)
+  var address,username;
+  //connect user to address if exists on the session
+  if (socket.request.session.user){
+    if (socket.request.session.user.address){
+      socket.join(socket.request.session.user.address.toString());
+    }
+  }
 
   socket.on('joinRoom', function(data){
-    var username = data.username;
-    var address = data.address;
-    //just binding values to socket for convenience
-    socket.username = username;
-    socket.chatRoom = address.toString();
+    username = socket.request.session.user.name;
+    //save new address to session
+    socket.request.session.user.address = data.address
+    socket.request.session.save();
+
+    console.log('joinRoom',socket.request.session)
+
+    address = socket.request.session.user.address;
     //join chat room with the name of the address
-    socket.join(socket.chatRoom);
-    console.log(socket.username,' joined room ',socket.chatRoom);
+    socket.join(address.toString());
+    console.log(username,' joined room ', address);
   });
 
   socket.on('leaveRoom', function(){
     socket.leaveRoom(socket.chatRoom);
-    console.log(socket.username,' left the room ',socket.chatRoom);
-    //remove chatRoom property
-    delete socket.chatRoom;
+    console.log(username,' left the room ',address);
+    //remove address property on session
+    delete address;
```

Comment:
this operator with a variable won't do anything; needs to operate on the property of an object